src/main/scala/viper/gobra/frontend/info/ExternalTypeInfo.scala

```diff
@@ -87,4 +87,7 @@ trait ExternalTypeInfo {
 
   /** returns all implementation proofs found in the current package */
   def localImplementationProofs: Vector[(Type, InterfaceT, Vector[String], Vector[String])]
+
+  /** returns the code root for a given node; can only be called on nodes that are enclosed in a code root */
+  def codeRoot(n: PNode): PCodeRoot with PScope
```

Comment:
We definitely could but this would mean yet another trait to add to most AST nodes. Even if we would do so, the type checker cannot give us any guarantees that `codeRoot` indeed does not cause an exception (we would have this assumption implicitly be carefully mixing the trait into AST nodes below code root